spec/datadog/core/telemetry/client_spec.rb

```diff
@@ -164,7 +165,8 @@
     let(:worker) { instance_double(Datadog::Core::Telemetry::Heartbeat) }
 
     before do
-      allow(Datadog::Core::Telemetry::Heartbeat).to receive(:new).and_return(worker)
+      allow(Datadog::Core::Telemetry::Heartbeat).to receive(:new)
+        .with(enabled: enabled, interval: heartbeat_interval).and_return(worker)
```

Comment:
**Note: This test was actually breaking because the let above was called `heartbeat_interval_seconds` and not `heartbeat_interval`.**

Minor: I suggest propagating the `_seconds` prefix internally as well:

```suggestion
      allow(Datadog::Core::Telemetry::Heartbeat).to receive(:new)
        .with(enabled: enabled, heartbeat_interval_seconds: heartbeat_interval_seconds).and_return(worker)
```